src/pages/index.js

```diff
@@ -74,115 +88,266 @@ const closeModalOnEsc = (event) => {
-  if (
-    openedModal &&
-    !openedModal
-      .querySelector(".modal__container, .modal__content")
-      .contains(event.target)
-  ) {
+  if (event.target.classList.contains("modal")) {
     closeModal(openedModal);
   }
 };
 
 // Event Listeners
-profileEditButton.addEventListener("click", () => {
-  profileNameInput.value = profileNameElement.textContent;
-  profileJobInput.value = profileJobElement.textContent;
-  openModal(editProfileModal);
+profileElements.editButton.addEventListener("click", () => {
+  profileElements.nameInput.value = profileElements.nameElement.textContent;
+  profileElements.jobInput.value = profileElements.jobElement.textContent;
+  openModal(profileElements.modal);
 });
 
-profileCloseButton.addEventListener("click", () => {
-  closeModal(editProfileModal);
+profileElements.closeButton.addEventListener("click", () => {
+  closeModal(profileElements.modal);
 });
 
-newPostButton.addEventListener("click", () => {
-  openModal(newPostModal);
+newPostElements.button.addEventListener("click", () => {
+  openModal(newPostElements.modal);
 });
 
-newPostCloseButton.addEventListener("click", () => {
-  closeModal(newPostModal);
+newPostElements.closeButton.addEventListener("click", () => {
+  closeModal(newPostElements.modal);
 });
 
-profileFormElement.addEventListener("submit", handleProfileFormSubmit);
+profileElements.form.addEventListener("submit", handleProfileFormSubmit);
+newPostElements.form.addEventListener("submit", handleNewPostFormSubmit);
+
+cardElements.modalCloseButton.addEventListener("click", () => {
+  closeModal(cardElements.modal);
+});
 
-newPostFormElement.addEventListener("submit", handleNewPostFormSubmit);
+cardElements.modalImage.addEventListener("click", () => {
+  closeModal(cardElements.modal);
+});
 
-// Form Submissions
+// Form Submission Handlers
 function handleProfileFormSubmit(event) {
   event.preventDefault();
 
-  profileNameElement.textContent = profileNameInput.value;
-  profileJobElement.textContent = profileJobInput.value;
+  const submitButton = profileElements.form.querySelector(
+    ".modal__submit-button"
+  );
+  const originalText = submitButton.textContent;
+  submitButton.textContent = "Saving...";
 
-  closeModal(editProfileModal);
+  api
+    .editUserInfo({
+      name: profileElements.nameInput.value,
+      about: profileElements.jobInput.value,
+    })
+    .then((data) => {
+      profileElements.nameElement.textContent = data.name;
+      profileElements.jobElement.textContent = data.about;
+      closeModal(profileElements.modal);
+    })
+    .catch((err) => {
+      console.error(`Error: ${err}`);
+    })
+    .finally(() => {
+      submitButton.textContent = originalText;
+    });
 }
 
 function handleNewPostFormSubmit(event) {
   event.preventDefault();
 
-  const newCardData = {
-    name: newPostprofileNameInput.value,
-    link: newPostLinkInput.value,
-  };
+  const submitButton = newPostElements.form.querySelector(
```

Comment:
**Could be improved**

  // the button is always available inside `event` as `submitter`
  const submitButton = event.submitter;